corehq/plugins/interface.py

```diff
@@ -11,36 +14,77 @@ class ExtensionPoint:
     docs = attr.ib(default="")
 
 
-class Plugins(ABC):
+class PluginContribution:
+    def __init__(self, callable_ref, domains=None):
+        self.callable_ref = callable_ref
+        self.domains = domains
+        self._callable = None
+
+    def validate(self, extension_point):
+        _callable = self.callable_ref
+        if isinstance(_callable, str):
+            _callable = to_function(self.callable_ref)
+        if not _callable or not callable(_callable):
+            raise Exception(f"Plugin callable is not callable: '{self.callable_ref}'")
```

Comment:
This module raises `Exception` in several places, which cannot safely be caught without possibly catching other unwanted exceptions. What do you think of adding a custom exception type to be used by this module?

Could be as simple as adding this at the bottom of this module, and then `raise Error(...)`
```py
class Error(Exception):
    pass
```